src/lurk/eval.rs

```diff
@@ -2023,65 +2027,120 @@ pub fn eval_let<F: AbstractField>() -> FuncE<F> {
     )
 }
 
-pub fn eval_letrec<F: AbstractField>() -> FuncE<F> {
+/// Tries to collect the bindings from a `letrec` (or `let`) expression. The bindings
+/// are turned into an environment for succinctness.
+/// In case of success, return `(1, collected_env)`. If an error `err` is found,
+/// return `(0, err)` instead.
+pub fn collect_bindings_as_an_env<F: AbstractField>() -> FuncE<F> {
     func!(
-        partial fn eval_letrec(binds_tag, binds, body_tag, body, env): [2] {
-            let err_tag = Tag::Err;
-            let invalid_form = EvalErr::InvalidForm;
+        fn collect_bindings_as_an_env(binds_tag, binds): [2] {
+            let zero = 0;
+            let invalid_form_err = EvalErr::InvalidForm;
             match binds_tag {
                 InternalTag::Nil => {
-                    let (res_tag, res) = call(eval_begin, body_tag, body, env);
-                    return (res_tag, res)
+                    let one = 1;
+                    return (one, zero)
```

Comment:
This is returning an empty environment right? That is, the zero here stands for the pointer to nil? Shouldn't we technically return `digests.lurk_symbol_ptr("nil")` here instead of `zero`? (I think they're the same (though now that I write it out I'm not so sure), but in other places we fetch the ptr like this instead of returning a literal zero)

Reply:
I'm returning zero because it's the value for the empty environment.